pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitTypeImpl.java

```diff
@@ -31,14 +38,57 @@ public class KitTypeImpl implements KitTypeDao {
             + "kit_type "
             + "WHERE kit_type_id = ?";
 
+    private static final String SQL_SELECT_KIT_TYPE = "SELECT * FROM ddp_kit_request_settings kitSettings "
+            + "LEFT JOIN kit_type type ON (type.kit_type_id = kitSettings.kit_type_id) "
+            + "LEFT JOIN ddp_instance realm ON (realm.ddp_instance_id = kitSettings.ddp_instance_id) "
+            + " WHERE realm.instance_name = ? AND type.kit_type_name = ?";
+
+    /**
+     * Getting kitTypes based on kit type name and realm
+     */
+    public static KitTypeDto getKitTypes(@NonNull String realm, @NonNull String kitTypeName) {
+        log.info("Finding kit typ for realm {} with kit type name {}", realm, kitTypeName);
+        List<KitTypeDto> kitTypes = new ArrayList<>();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_KIT_TYPE)) {
+                stmt.setObject(1, realm);
+                stmt.setObject(2, kitTypeName);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        kitTypes.add(KitTypeDto.builder()
+                                .withKitTypeId(rs.getInt(DBConstants.KIT_TYPE_ID))
+                                .withKitTypeName(rs.getString(DBConstants.KIT_TYPE_NAME))
+                                .withDisplayName(rs.getString(DBConstants.DISPLAY_NAME))
+                                .build());
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Error looking up kit type  ", results.resultException);
+        }
+        if (kitTypes.size() != 1) {
+            // JDBC does not fetch all the data at once, so we need to read all the kit types here and then later
+            // throw an error if there was more than one kit type for this realm with the same name
```

Comment:
I don't understand this comment. The test here is pretty clear that we are expecting only one result. If so, it probably makes sense to handle the results differently when processing the query,